### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by ptbrowne**
## Goals

Upgrade next and React
Remove theme-ui and use MUI as in other gov projects
Also removed public/docs which contained an outdated invision prototype that was no longer accessible

## Comments

- Could remove downshift and reuse Autocomplete from MUI
- Could remove reach packages that were used for the Dialog
- Not to deploy now as we have problems to deploy on CloudFoundry with nextjs > 11, let's wait until we change architecture, at the end of the year

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,15 +69,15 @@
     "make-plural": "^7.0.0",
     "micromark": "^2.11.2",
     "micromark-extension-gfm": "^0.3.3",
-    "next": "^11.0.1",
+    "next": "14",
```
Comment (bprusinowski) [inline package.json:72]: Wondering, why not a more specific version? 😄

File: src/components/charts-generic/interaction/ruler.tsx [hunk 2]
```diff
@@ -64,7 +64,7 @@ export const RulerContent = ({
           position: "absolute",
           borderWidth: 0.5,
           borderStyle: "solid",
-          borderColor: "monochrome.200",
+          borderColor: "grey[].200",
```
Comment (bprusinowski) [inline src/components/charts-generic/interaction/ruler.tsx:67]: I think it should be `grey[200]`?

File: src/components/combobox.tsx [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import { t } from "@lingui/macro";
 // import { i18n } from "@lingui/core";
```
Comment (bprusinowski) [inline src/components/combobox.tsx:2]: To remove?

File: src/themes/elcom.ts [hunk 4]
```diff
@@ -21,7 +22,72 @@ const grey = {
   900: "#000000",
 } as const;
 
-export const theme: Theme = createTheme({
+const breakpoints = ["xs", "md", "md"] as Breakpoint[];
```
Comment (bprusinowski) [inline src/themes/elcom.ts:31]: Just a comment, why only three, and why md twice? 😅 I see that down below we have xs, sm, md lg and xl?

File: src/components/VisuallyHidden.tsx [hunk 5]
```diff
@@ -2,7 +2,7 @@ import { Box } from "@mui/material";
 import { visuallyHidden } from "@mui/utils";
 
 const VisuallyHidden = ({ children }: { children: React.ReactNode }) => (
-  <Box sx={visuallyHidden}>{children}</Box>
+  <Box style={visuallyHidden}>{children}</Box>
```
Comment (bprusinowski) [inline src/components/VisuallyHidden.tsx:5]: Super-extra-nit: If it's a `style` now, you can also remove `Box` 🥹

File: src/components/form.tsx [hunk 6]
```diff
@@ -114,25 +118,25 @@ export const Checkbox = ({
...
 }: { label: React.ReactNode; disabled?: boolean } & FieldProps) => (
-  <Box
-    component="label"
+  <FormControlLabel
     label={label}
     htmlFor={`${name}-${label}`}
+    control={
+      <MuiCheckbox
+        sx={{
+          // size: 20,
```
Comment (bprusinowski) [inline src/components/form.tsx:127]: Nit: Remove comment?

File: src/components/logo.tsx [hunk 7]
```diff
@@ -1,11 +1,12 @@
...
 export const LogoMobile = () => (
   <svg width={30} height={34}>
     <title id="logo">
-      <Trans id="logo.swiss.confederation">
-        Logo of the Swiss Confederation
-      </Trans>
+      {t({
+        id: "logo.swiss.confederation",
+        message: "Logo of the Swiss Confederation",
+      })}
```
Comment (bprusinowski) [inline src/components/logo.tsx:9]: Wondering why it was a problem, and why it was not a problem e.g. in `LogoDesktop` down below? 🤔

File: src/components/detail-page/selector-multi.tsx [hunk 8]
```diff
@@ -50,7 +50,6 @@ export const SelectorMulti = ({
         variant="lead"
         sx={{ display: "contents" }}
       >
-        helo
         <Trans id="selector.legend.select.parameters">
```
Comment (bprusinowski) [inline src/components/detail-page/selector-multi.tsx]: 😅

File: src/components/logo.tsx [hunk 9]
```diff
@@ -24,9 +24,10 @@ export const LogoMobile = () => (
 export const LogoDesktop = () => (
   <svg width={224} height={56}>
     <title id="logo">
-      <Trans id="logo.swiss.confederation">
-        Logo of the Swiss Confederation
-      </Trans>
+      {t({
+        id: "logo.swiss.confederation",
+        message: "Logo of the Swiss Confederation",
+      })}
```
Comment (bprusinowski) [inline src/components/logo.tsx:30]: Ok, now I see it was also a problem 😅

File: src/components/search.tsx [hunk 10]
```diff
@@ -323,7 +323,3 @@ const getEntity = (e: ResultType) => {
       "operator";
   }
 };
-
-const removeDiacritics = (str: string) => {
-  return str.normalize("NFD").replace(/[\u0300-\u036f]/g, "");
-};
```
Comment (bprusinowski) [inline src/components/search.tsx]: So it's not needed to remove diacritics anymore, or is it handled in some other way?

**Review by bprusinowski — APPROVED** [review]
Code LGTM, with mostly some questions 👏

I did a brief visual review and found that the chevron is too close to the text in a screenshot below. I will try to have a more detailed look by the end of Monday 👍 

Screenshot 2024-09-13 at 15 25 39

File: src/components/search.tsx (hunk 10 above)
Comment (ptbrowne) [reply src/components/search.tsx]: It is handled by the Autocomplete component, see https://mui.com/material-ui/react-autocomplete/#custom-filter 

> config.ignoreAccents (bool [optional]): Defaults to true. Remove diacritics.

Comment (bprusinowski) [reply src/components/search.tsx]: Cool, thanks for explaining 👍

File: src/themes/elcom.ts (hunk 4 above)
Comment (ptbrowne) [reply src/themes/elcom.ts]: These were the breakpoints for the font specs. The extra md was an error. I auditted the values for the fontSpecs and have seen that it was always the same size going up, in fact there were only two sizes (mobile and desktop). I tweaked the code in https://github.com/visualize-admin/electricity-prices-switzerland/pull/78/commits/c6cc46a114ea2907eaa46aa6144247aa1d5b60e0. Thanks for noticing.

**Comment by ptbrowne** [comment]
@bprusinowski I have merged this branch into the next branch.
The updates on Next were not working on CloudFoundry but we will soon change environment.
This is why, I keep main without the updates in case we need to do hotfixes on main on CloudFoundry.
So `next` is the new `main`. `main` should receive bug fixes and other small fixes until we deploy on the new platform. Any fixes that are done on `main` should also be done on `next`.

**Pull request merged**